### PR TITLE
DES-130: search for a resident

### PR DESCRIPTION
**Pull request by neilmendum**
https://hackney.atlassian.net/browse/DES-130

- Search for a resident
- Display loading table while results are loading
- Documented my understanding of the different implementations in `gateways`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -103,7 +103,28 @@ You can use a `.env` file to supply environment config locally. Create a fresh o
 | EVIDENCE_API_TOKEN_DOCUMENT_TYPES_GET     |                                                                     |                                     |
 | EVIDENCE_API_TOKEN_EVIDENCE_REQUESTS_GET  |                                                                     |                                     |
 | EVIDENCE_API_TOKEN_EVIDENCE_REQUESTS_POST |                                                                     |                                     |
+| EVIDENCE_API_TOKEN_RESIDENTS_GET          |                                                                     |                                     |
```
Comment (jhackett1) [inline README.md:106]: are all these tokens definitely going to be different?

**Review by jhackett1 — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (neilmendum) [reply README.md:106]: I believe so yes - each HTTP verb for a given path has unique tokens created

File: README.md [hunk 2]
```diff
@@ -103,7 +103,28 @@ You can use a `.env` file to supply environment config locally. Create a fresh o
...
+
+Following [Hackney's API Playbook](https://github.com/LBHackney-IT/API-Playbook-v2-beta) we have a `Gateways` directory which:
+
+> Holds the class responsible for establishing connection with the data source and retrieving/inserting/updating data queries to perform the given action against the data source
+
+At the time of writing we have two Gateways which are for specific use cases:
+
+- **`internal-api.ts`**
+
+  - This acts a means of routing client side requests, for example form submissions, to a proxy endpoint on the Next.js server.
```
Comment (bogdan-zaharia) [inline README.md:122]: I think there might be a typo here

**Review by bogdan-zaharia — DISMISSED** [review]
LGTM!

File: README.md (hunk 2 above)
Comment (neilmendum) [reply README.md]: Great spot 👍🏻  fixed

**Review by bogdan-zaharia — APPROVED** [review]

**Pull request merged**